QuizApp/Quiz/DataSource/QuizDataSource.swift

```diff
@@ -14,14 +14,24 @@ class QuizDataSource: QuizDataSourceProtocol {
         self.quizClient = quizClient
     }
 
+    @MainActor
     func getQuizzes(for category: String) async throws -> [QuizDataModel] {
-        return try await quizClient.getQuizzes(for: category)
-            .map { QuizDataModel(from: $0) }
+        do {
+            let quizzes = try await quizClient.getQuizzes(for: category).map { QuizDataModel(from: $0) }
+            return quizzes
+        } catch let error {
+            throw error
+        }
```

Comment:
No need to add do-catch if the catch block is only throwing the same error it received. 
In this case 
```
return try await quizClient.getQuizzes(for: category)
            .map { QuizDataModel(from: $0) }
```

is exactly the same as: 

```
        do {
            let quizzes = try await quizClient.getQuizzes(for: category).map { QuizDataModel(from: $0) }
            return quizzes
        } catch let error {
            throw error
        }
```
only longer. So the first option is preferred.